docs/resources/control-flow/functions/action-triggers.md

```diff
@@ -0,0 +1,291 @@
+    height: 0,
+    width: '100%'}}>
+    <iframe 
+        src="https://demo.arcade.software/uI8KTjnmre4Cx901b6ws?embed&show_copy_link=true"
+        title=""
+        style={{
+            position: 'absolute',
+            top: 0,
+            left: 0,
+            width: '100%',
+            height: '100%',
+            colorScheme: 'light'
+        }}
+        frameborder="0"
+        loading="lazy"
+        webkitAllowFullScreen
+        mozAllowFullScreen
+        allowFullScreen
+        allow="clipboard-write">
+    </iframe>
+</div>
+<p></p>
+
+### Widget Specific Triggers
+
+Widget specific triggers are available on certain widgets. Here are they:
+
+- **On Submit**: Triggered on the TextField widget when the user presses "submit" or "done," finalizing text entry.
+- **On Page Load**: Available on the page widget, this trigger activates as soon as the page loads, useful for tasks like data fetching or content updates.
+- **On Phone Shake**: Specific to the page widget, this trigger responds to physical shaking of the device, commonly used in games for actions like rolling dice.
+- **On Selected**: Found on widgets like Dropdowns, CheckboxGroups, Sliders, RadioButtons, ChoiceChips, and RatingBars, this trigger activates upon any change in selection.
+- **On Page Swipe**: Available on the PageView widget to trigger actions when the page is swiped.
+- **On Toggle**: Available on the ToggleIcon widget, this trigger responds each time the toggle is activated.
+- **On Completed, On Change**: Specific to the PinCode widget, these triggers activate when the user completes or alters a pin entry.
+- **On Count Changed**: Present in the CountController widget, this trigger responds to changes in the count.
+
+## Gesture Detector Triggers
+
+Gesture Detector Triggers enable you to respond to user gestures, such as taps, drags, swipes, and pinches. These triggers are invoked based on specific gestures and allow you to add actions in response to user interactions. For example, the `onDoubleTap` trigger is invoked whenever a user quickly taps twice on a widget, which can be used to toggle a 'like' state or zoom in on content.
+
+### Lifecycle stages
+
+The lifecycle of gesture triggers involves four key stages: **Start**, **Update**, **End/Stop**, and **Cancel**. These stages dictate how gestures are detected and handled, from the initial interaction to completion or cancellation. Understanding this lifecycle is crucial for building intuitive gesture-based interactions in your app.
+
+#### Tap Gesture Lifecycle
+
+Tap gestures have a simpler lifecycle, focusing primarily on detecting taps and whether they complete or get canceled. Here's how the tap lifecycle works:
+
+1. **Start**: This stage begins when the user places their finger on the screen to initiate a tap. For example: `onTapDown` is triggered when the user touches the screen to start a tap.
```

Comment:
So Start, End made sense for drag gestures because all of the gestures are named similarly, why do we need to use the same terminology for this? Why not use Down, Up, Cancel terminology directly? Or maybe we use Start/Down, End/Up, Cancel instead? 

This applies to the diagram too. And it would be great to showcase that onTap is called after onTapUp (in the diagram)

Reply:
Ya, no worries. updated!